src/analysis/calculus/times_cont_diff.lean

```diff
@@ -66,16 +66,29 @@ One could define `C^n` functions in a domain `s` by fixing an arbitrary choice o
 is what we do with `iterated_fderiv_within`) and requiring that all these derivatives up to `n` are
 continuous. If the derivative is not unique, this could lead to strange behavior like two `C^n`
 functions `f` and `g` on `s` whose sum is not `C^n`. A better definition is thus to say that a
-function is `C^n` inside `s` if it admits a sequence of derivatives up to `n` inside `s`. This
-definition still has the problem that a function which is locally `C^n` would not need to be `C^n`,
-as different choices of sequences of derivatives around different points might possibly not be glued
-together to give a globally defined sequence of derivatives. Also, there are locality problems in
-time: one could image a function which, for each `n`, has a nice sequence of derivatives up to order
-`n`, but they do not coincide for varying `n` and can therefore not be glued to give rise to an
-infinite sequence of derivatives. This would give a function which is `C^n` for all `n`, but not
-`C^∞`. We solve this issue by putting locality conditions in space and time in our definition of
-`times_cont_diff_on`. The resulting definition is slightly more complicated to work with (in fact
-not so much), but it gives rise to completely satisfactory theorems.
+function is `C^n` inside `s` if it admits a sequence of derivatives up to `n` inside `s`.
+
+This definition still has the problem that a function which is locally `C^n` would not need to
+be `C^n`, as different choices of sequences of derivatives around different points might possibly
+not be glued together to give a globally defined sequence of derivatives. (Note that this issue
+can not happen over reals, thanks to partition of unity, but the behavior over a general field is
```

Comment:
This parenthesis explains a lot of my confusion. I think it should be moved up, clearly stating from the beginning that part of the complexity of the coming explanation and Lean code stems from supported normed fields more general than reals and complexes. It would still be nice to have a pathological example showing the difficulty but this is much less important if it doesn't happen with real numbers.